stackslib/src/net/chat.rs

```diff
@@ -2687,6 +2699,155 @@ impl ConversationP2P {
         }
     }
 
+    /// Are we trying to resolve DNS?
+    pub fn waiting_for_dns(&self) -> bool {
+        self.dns_deadline < u128::MAX
+    }
+
+    /// Try to get the IPv4 or IPv6 address out of a data URL.
+    fn try_decode_data_url_ipaddr(data_url: &UrlString) -> Option<SocketAddr> {
+        // need to begin resolution
+        // NOTE: should always succeed, since a UrlString shouldn't decode unless it's a valid URL or the empty string
+        let Ok(url) = data_url.parse_to_block_url() else {
```

Comment:
NIT: You can also replace this with 
```let url = data_url.parse_to_block_url().ok()?;``` since the function returns an optional.

Reply:
sure